### ISSUE TITLE
evil-window-move-far-* commands cannot handle 'no-delete-other-windows window parameter

### ISSUE THREAD

**Issue by Alexander-Miller** [emacs-evil/evil#1129]
# Issue type

- Bug report

# Environment

Emacs version: GNU Emacs 26.1 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.22.30) of 2018-07-05
Operating System: Linux Antergos
Evil version: evil-git-a2bda1682
Evil installation type: MELPA
Graphical/Terminal: GUI
Tested in a `make emacs` session (see [CONTRIBUTING.md](https://github.com/emacs-evil/evil/blob/master/CONTRIBUTING.md#bug-reports)): Tested with `emacs -Q`

# Reproduction steps
 - emacs -Q
 - run this:
```
(setq package-user-dir "/home/a/.emacs.d/elpa/26.1/develop/")
(package-initialize)
(require 'evil)

(with-selected-window (display-buffer-in-side-window
                       (get-buffer-create "TEST")
                       '((side . left)))
  (set-window-dedicated-p (selected-window) t)
  (set-window-parameter (selected-window) 'no-delete-other-windows t))
```
- `M-x evil-window-move-far-right`

# Expected behavior
The selected window is moved to the far right, the TEST window remains on the left.

# Actual behavior
https://imgur.com/a/IbHRiml

# Further notes
More details in [this comment](https://github.com/Alexander-Miller/treemacs/issues/349#issuecomment-463257206).

**Comment by dzop** [comment]
I was similarly annoyed that the window move commands can't handle window parameters. The problem is that the functions that those commands use only look at the buffers of the windows and don't consider window parameters. Specifically `evil-get-buffer-tree` and `evil-restore-window-tree`.

I was able to solve this by using `window-state-get` and `window-state-put`. I just submitted a [pull request](https://github.com/emacs-evil/evil/pull/1130) which may solve your problem.
